### PR TITLE
fix: nameet validation updation on driver and carrier onboard

### PR DESCRIPTION
**Pull request by nameetjambla**
In this file, I have solved GitHub issue number 1081 which is related to carrier onboard validations and also updated the validation of the driver module.

Kindly Ignore this file this is not the updated file

### CODE REVIEW

File: src/app/entry/onboard/add-account/add-account.component.html [hunk 1]
```diff
@@ -521,10 +521,10 @@ <h2 class="mt-1">Create Your Fleet Hawks Account</h2>
                             <div class="col-lg-10">
                               <label class="control-label font-weight-bold font-weight-bold text-lg-right pt-2">Postal Code</label>
                               <input [(ngModel)]="item.zipCode" name="zipCode{{i}}" type="text" class="form-control"
-                                placeholder="Enter zipode eg. 7V2 3R4"  #zip="ngModel" pattern="^[A-Z0-9]{6}$">
+                                placeholder="Enter zipode eg. 7V2 3R4"  #zip="ngModel" pattern="^[a-zA-Z0-9\s\#\.\'\-\/\,]{2,7}$">
```
Comment (Hardeep8587) [inline src/app/entry/onboard/add-account/add-account.component.html]: what is this zip code? message is something, pattern is something

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 2]
```diff
@@ -649,8 +663,8 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                               Document# is required.
                             </div>
                             <div *ngIf="docNum.errors.pattern">
-                              Document# must contain combination of small letters,capital letters
-                              ,alphanumeric and atleast one special characters(#,/,-,_,.)
+                              Document# must contains alphanumeric and special characters for period(space,#,-,_,.) only.
```
Comment (Hardeep8587) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: very much confusing message, period means (.) - dot

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 3]
```diff
@@ -1161,36 +1250,64 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                             Miles<span class="mandfield text-2 ml-1"><sup>*</sup></span></label>
                           <input [(ngModel)]="driverData.paymentDetails.loadedMiles" name="paymentDetails.loadedMiles"
 
-                            type="text" class="form-control" placeholder="eg. 10,20">
+                            type="text" class="form-control" placeholder="eg. 10,20" #lM="ngModel" required pattern="^[0-9]+$">
+                            <div *ngIf="lM.invalid && (lM.dirty || lM.touched)" class="text-danger">
+                              <div *ngIf="lM.errors.required">
+                               Loaded is required.
```
Comment (Hardeep8587) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: messages are not very clear. Loaded mile rate is required.

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 4]
```diff
@@ -1199,17 +1316,31 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                           <input [(ngModel)]="driverData.paymentDetails.loadedMilesTeam"
                             name="paymentDetails.loadedMilesTeam" type="text" class="form-control"
 
-                            placeholder="eg. 10,20">
+                            placeholder="eg. 10,20" #lM2="ngModel" required pattern="^[0-9]+$">
+                            <div *ngIf="lM2.invalid && (lM2.dirty || lM2.touched)" class="text-danger">
+                              <div *ngIf="lM2.errors.required">
+                              Loaded Miles(Team) is required.
```
Comment (Hardeep8587) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html:1322]: loaded miles team is required? where is it ?

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 5]
```diff
@@ -1218,17 +1349,32 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                           <input [(ngModel)]="driverData.paymentDetails.emptyMilesTeam"
                             name="paymentDetails.emptyMilesTeam" type="text" class="form-control"
 
-                            placeholder="eg. 10,20">
+                            placeholder="eg. 10,20" #eM3="ngModel" required pattern="^[0-9]+$">
+                            <div *ngIf="eM3.invalid && (eM3.dirty || eM3.touched)" class="text-danger">
+                              <div *ngIf="eM3.errors.required">
```
Comment (Hardeep8587) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: please check thoroughly

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 6]
```diff
@@ -1377,10 +1529,16 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                         </label>
                         <ng-select [(ngModel)]="driverData.hosDetails.hosCycle" name="hosDetails.hosCycle"
                           (change)="onChangeHideErrors('hosDetails.hosCycle')" class="form-control populate"
-                          placeholder="Select the cycle for the driver">
+                          placeholder="Select the cycle for the driver" #sC="ngModel" required>
                           <ng-option class="text-capitalize" *ngFor="let cycle of cycles" value="{{cycle.cycleID}}">
                             {{cycle.cycleName}}</ng-option>
                         </ng-select>
+                        <div *ngIf="sC.invalid && (sC.dirty || sC.touched)" class="text-danger">
```
Comment (Hardeep8587) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html:1536]: cycle is required when it is non exempted, please add condition on required, u can refer to bizcountry condition in onboard form

**Review by Hardeep8587 — COMMENTED** [review]
please check thoroughly all the messages, validations.

File: src/app/entry/onboard/add-account/add-account.component.html (hunk 1 above)
Comment (nameetjambla) [reply src/app/entry/onboard/add-account/add-account.component.html]: solved

**Review by nameetjambla — COMMENTED** [review]
@Hardeep8587 mam, I Have done all the changes except the last review I just want to know that change should be done on the add-driver ts file?

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 2 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: solved

**Review by FH-sandeepkaur1 — APPROVED** [review]

**Comment by kunalfleethawks** [comment]
@nameetjambla  Please resolve conflicts

**Review by kunalfleethawks — APPROVED** [review]

**Comment by nameetjambla** [comment]
this file consist of old code  and not updated with vaidations

**Pull request closed without merging**